File: flutter_food_finder/lib/screens/favourites_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_food_finder/providers/favourites_provider.dart';
import 'package:flutter_food_finder/widgets/restaurant_list.dart';
import 'package:provider/provider.dart';

class FavouritesScreen extends StatelessWidget {
  const FavouritesScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final restaurants = context.watch<FavouritesProvider>().favourites;

    return Scaffold(
        appBar: AppBar(title: const Text("Favourites")),
        body: RestaurantList(restaurants: restaurants));
  }
}

File: react-food-finder/src/screens/FavouritesScreen.js
import React, { useContext } from 'react';

import { FavouritesContext } from '../context/FavouritesContext';
import { LocationContext } from '../context/LocationContext';

import Container from '../components/Container';
import RestaurantList from '../components/RestaurantList';

const FavouritesScreen = () => {
  const { favourites } = useContext(FavouritesContext);
  const { location } = useContext(LocationContext);

  return (
    <Container>
      {location && <RestaurantList restaurants={favourites} />}
    </Container>
  );
};

export default FavouritesScreen;
